### ISSUE TITLE
App Freeze or Hang

### ISSUE THREAD

**Issue by ghost** [mirsamantajbakhsh/TorAndroid#3]
When I call your Method then I check on Android studio log everything is fine tor connected successfully, but my application is hang on or freeze please help me with this

**Comment by mirsamantajbakhsh** [comment]
Hi.

Try start it in a background thread. Not UI thread.

**Comment by ghost** [comment]
Can You give me sample code

**Comment by mirsamantajbakhsh** [comment]
Sample code of connection is available in README.md and AndroidTest.
https://github.com/mirsamantajbakhsh/TorAndroid/blob/main/app/src/androidTest/java/ir/mstajbakhsh/torandroid/ExampleInstrumentedTest.java

Just put that code inside an AsynchTask
https://developer.android.com/reference/android/os/AsyncTask

**Comment by ghost** [comment]
Can you Integrate My Project I will hire you

**Comment by mirsamantajbakhsh** [comment]
I can check your code. Upload it to github and send me the repository link, but for hiring, I am busy and cannot be hired. Thanks for suggestion.

**Comment by ghost** [comment]
https://we.tl/t-Iu3BddLtKL

check this code

**Comment by mirsamantajbakhsh** [comment]
It did not load. Would you please paste the code here as a code segment like:
```sample code segment```?

**Comment by ghost** [comment]
[Lightning-Browser.zip](https://github.com/mirsamantajbakhsh/TorAndroid/files/6344705/Lightning-Browser.zip)

**Comment by mirsamantajbakhsh** [comment]
I've checked the code. You have used the TorAndroid library in lightning/connect/TorConnect.java file. I've tried your code and faced an exception in library and created an issue #4 and fixed it in 3a48012bf6e09da622c97bf426df559b0e2fa9ec

I've uploaded the library to Bintray and now it is working. Here is a sample code which will start Tor in another thread. Feel free to ask if you have another question.

```java
    public void startTorInBackground(Context cntx) {
        HandlerThread mHandlerThread = new HandlerThread("TorThread");
        mHandlerThread.start();
        Handler mHandler = new Handler(mHandlerThread.getLooper());
        mHandler.postDelayed(() -> {
            final TorProxy tb = new TorProxy.TorBuilder()
                    .setSOCKsPort(9150)
                    .setUseBrideges(false)
                    .setDebuggable(false)
                    .build(cntx.getApplicationContext());
            tb.init();
            try {
                final IConnectionDone icd = new IConnectionDone() {
                    @Override
                    public void onSuccess() {
                        Log.d("TorAndroid", "Tor Started Successfully.");
                    }

                    @Override
                    public void onFailure(Exception ex) {
                        Log.e("TorAndroid", "Error in Starting Tor.\r\n" + ex.getMessage());
                    }
                };
                tb.start(icd);
            } catch (IOException e) {
                e.printStackTrace();
            }
        }, 1000);
    }
```

**Comment by ghost** [comment]
I integrate this but when I try to use onion URL in my browser application then not working also https://check.torproject.org open not connected, please check yourself

**Comment by ghost** [comment]
> ```java
> public void startTorInBackground(Context cntx) {
>         HandlerThread mHandlerThread = new HandlerThread("TorThread");
>         mHandlerThread.start();
>         Handler mHandler = new Handler(mHandlerThread.getLooper());
>         mHandler.postDelayed(() -> {
>             final TorProxy tb = new TorProxy.TorBuilder()
>                     .setSOCKsPort(9150)
>                     .setUseBrideges(false)
>                     .setDebuggable(false)
>                     .build(cntx.getApplicationContext());
>             tb.init();
>             try {
>                 final IConnectionDone icd = new IConnectionDone() {
>                     @Override
>                     public void onSuccess() {
>                         Log.d("TorAndroid", "Tor Started Successfully.");
>                     }
> 
>                     @Override
>                     public void onFailure(Exception ex) {
>                         Log.e("TorAndroid", "Error in Starting Tor.\r\n" + ex.getMessage());
>                     }
>                 };
>                 tb.start(icd);
>             } catch (IOException e) {
>                 e.printStackTrace();
>             }
>         }, 1000);
>     }
> ```

This Code not working, Nullpointer exception

**Comment by mirsamantajbakhsh** [comment]
I think that you are using un-updated version of the library (May be the old version is in your Gradle's cache). However, I've released a new version both in GitHub release page and Bintray.

Update your library version to 2.0.0447 and it will be fixed.

```
implementation 'com.github.mirsamantajbakhsh:TorAndroid:2.0.0447'
```

I've checked it myself. Now it is working.

![image](https://user-images.githubusercontent.com/10776156/115956058-f3e69700-a50f-11eb-9352-800748ba8c67.png)

I've used the same code I've sent previously:
![image](https://user-images.githubusercontent.com/10776156/115956153-78391a00-a510-11eb-836b-1bc6cea20666.png)
